ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/PluginGeneralInfoSection.java

```diff
@@ -74,6 +74,7 @@ protected void createSpecificControls(Composite parent, FormToolkit toolkit, IAc
 		if (isBundle() && (formEditor instanceof ManifestEditor)) {
 			createLazyStart(parent, toolkit, actionBars);
 			createSingleton(parent, toolkit, actionBars, PDEUIMessages.PluginGeneralInfoSection_singleton);
+			createBundleShape(parent, toolkit, PDEUIMessages.PluginGeneralInfoSection_bundleshape);
```

Comment:
By calling `createBundleShape()` here the Bundle-Shape buttons are only available for non-fragment bundles, but fragments can also have an explicit Bundle-Shape.
So this should called for all kind of Plug-in projects in a way that it is still at the right position.
So probably in `GeneralInfoSection.createClient()` after `createSpecificControls()` has been called.

Reply:
Hi @HannesWell ,
I have addressed all the review comments. Please have a look and let me know. (I am unable to select the checkboxes in the above comment.)

Reply:
Thank you. I'm a bit short in time today, but will have a look at this ASAP.
But I think we can submit this soon. :)